librz/core/disasm.c

```diff
@@ -5292,17 +5365,19 @@ RZ_API int rz_core_print_disasm(RzPrint *p, RzCore *core, ut64 addr, ut8 *buf, i
 
 	// TODO: All those ds must be print flags
 	RzDisasmState *ds = ds_init(core);
-	ds->cbytes = cbytes;
+	ds->cbytes = options ? options->cbytes : 0;
 	ds->print = p;
-	ds->l = l;
+	ds->l = nlines;
 	ds->buf = buf;
 	ds->len = len;
 	ds->addr = addr;
 	ds->hint = NULL;
 	ds->buf_line_begin = 0;
-	ds->pdf = pdf;
+	ds->pdf = options ? options->function : NULL;
+	ds->pj = NULL;
+	ds->vec = options ? options->vec : NULL;
```

Comment:
Again, I think this is not alright. `ds->vec` should be initialized with a new Rzvector imho because you are asking `rz_core_print_disasm` to disassemble some instructions and the result will be in `->vec`, thus the vec will be filled from the function and I don't see reasons to append stuff to existing instructions.

Reply:
Just because our cutter uses the `pdJ` command,
So I want to get this data on the caller side instead of a string in JSON format.

```
[0x00401cf0]> pdJ 3 | jq
[
  {
    "offset": 4201712,
    "text": "        \u001b[34m╎\u001b[0m   \u001b[36m;-- entry0:\u001b[0m"
  },
  {
    "offset": 4201712,
    "text": "        \u001b[34m╎\u001b[0m   \u001b[32m0x00401cf0\u001b[0m      \u001b[1;92mcall  0x401fdb\u001b[0m\u001b[0m"
  },
  {
    "offset": 4201717,
    "arrow": 4201161,
    "text": "        \u001b[34m└\u001b[0m\u001b[34m─\u001b[0m\u001b[34m<\u001b[0m \u001b[32m0x00401cf5\u001b[0m      \u001b[32mjmp   0x401ac9\u001b[0m\u001b[0m"
  },
  {
    "offset": 4201722,
    "text": "            \u001b[32m0x00401cfa\u001b[0m      \u001b[1;91mint3\u001b[0m\u001b[0m\u001b[0m"
  }
]
```

Reply:
Yes, I agree, but it should be something like
```
ds->vec = rz_list_newf(....);
```
instead of what it is now.

Reply:
But if I do this, I can only get this data on the caller by using `ds->vec` as the return value, and then this breaks the current return value system.

And now, I just need

```c
RzPVector* vec = rz_pvector_new(...)
rz_core_print_disasm(...vec)
```

If I do not need the data

```c
rz_core_print_disasm(...vec:NULL)
```